### PR TITLE
[Proof] Improve arb_accumulator_proof

### PR DESCRIPTION
**Pull request by wqfish**
## Motivation

The existing way of implementing `Arbitrary` is not very nice. Do it properly.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Probably.

## Test Plan

Existing tests.

## Related PRs

None.

### CODE REVIEW

File: types/src/proof/proptest_proof.rs [hunk 1]
```diff
@@ -15,20 +15,26 @@ use crypto::{
...
 use rand::{seq::SliceRandom, thread_rng};
 
+fn arb_non_placeholder_accumulator_sibling() -> impl Strategy<Value = HashValue> {
+    any::<HashValue>().prop_filter("Filter out placeholder sibling.", |x| {
+        *x != *ACCUMULATOR_PLACEHOLDER_HASH
+    })
+}
+
+fn arb_accumulator_sibling() -> impl Strategy<Value = HashValue> {
+    prop_oneof![
```
Comment (huitseeker) [inline types/src/proof/proptest_proof.rs]: Do you want to use the [weighed variant](https://docs.rs/proptest/0.9.4/proptest/macro.prop_oneof.html) of `oneof` here?

Comment (wqfish) [reply types/src/proof/proptest_proof.rs]: I think uniform weight is okay here...

**Pull request closed without merging**